src/components/create-modal-button/index.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import kebabCase from 'lodash/kebabCase';
+
+import Button from '../button';
+import Icon from '../icon';
+import ModalPortal from '../modal-portal';
+import createLazyComponent from '../createLazyComponent';
+
+export default class CreateModalButton extends React.Component {
+  static propTypes = {
+    icon: PropTypes.string,
+    iconUrl: PropTypes.string,
+    modal: PropTypes.objectOf(Promise).isRequired,
+    onShow: PropTypes.func,
+    onHide: PropTypes.func,
+    text: PropTypes.string,
+  };
+
+  static defaultProps = {
+    icon: '',
+    iconUrl: '',
+    onShow: undefined,
+    onHide: undefined,
+    text: 'Create',
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showModal: false,
+    };
+    this.handleShowModal = this.handleShowModal.bind(this);
+    this.handleHideModal = this.handleHideModal.bind(this);
+  }
+
+  handleShowModal() {
```

Comment:
think you can reduce this to a single toggle function

```
  toggle() {
    const { onHide, onShow } = this.props;
    this.setState({
      showModal: !this.state.showModal,
    }, () => {
      if (this.state.showModal) {
        onShow();
      } else {
        onHide();
      }
    });
  }
```

im also not convinced there needs to be onShow/onHide. why not just provide a toggle prop and call that toggle prop with the new value? or not have onShow/onHide at all... what does that allow us to do that we wouldnt otherwise be able to do? if it is going to be torn down and remounted each time then we can use the modal's lifecycle methods

Reply:
You are correct. I was totally confused by the examples on react portals and forgot this will actually renders a reactstrap Modal. (This same reason I created the ModalPortal, I've already removed it. Thanks.) All I want to do here is give the button some ability to run some code with the data maybe only accessible from where the btn rendered. But that's can be done via the onClick or the modal lifecycle with some props.